llrt_core/src/modules/encoding/text_encoder.rs

```diff
@@ -21,4 +24,34 @@ impl TextEncoder {
     pub fn encode<'js>(&self, ctx: Ctx<'js>, string: String) -> Result<Value<'js>> {
         TypedArray::new(ctx, string.as_bytes()).map(|m| m.into_value())
     }
+
+    pub fn encode_into<'js>(
+        &self,
+        ctx: Ctx<'js>,
+        string: String,
+        obj: Object<'js>,
+    ) -> Result<Object<'js>> {
+        let mut read = 0;
+        let mut written = 0;
+
+        if let Some((array_buffer, source_length, source_offset)) = obj_to_array_buffer(&obj)? {
+            let raw = array_buffer
```

Comment:
I don't think you need to access the raw bytes like this.
You can do `let bytes: &[u8] = array_buffer.as_ref()` instead and use offsets on slice instead.

Reply:
The type of bytes must be `&mut [u8]`, not `&[u8]`.
I could not find a way to create `&mut [u8]` other than the current implementation...

Reply:
Ah right, makes sense since we're writing into the typed array!